Tests/FuzzilliTests/ProgramBuilderTest.swift

```diff
@@ -184,6 +184,119 @@ class ProgramBuilderTests: XCTestCase {
+        let _ = b.defineClass(withSuperclass: superclass) { cls in
+            cls.defineConstructor(withParameters: [.string]) { params in
+                let v3 = b.loadInt(0)
+                let v4 = b.loadInt(2)
+                let v5 = b.loadInt(1)
+                b.forLoop(v3, .lessThan, v4, .Add, v5) { _ in
+                    let v0 = b.loadInt(42)
+                    let v1 = b.createObject(with: ["foo": v0])
+                    b.callSuperConstructor(withArgs: [v1])
+                }
+            }
+
+            cls.defineProperty("b")
+
+            cls.defineMethod("g", withSignature: [.anything] => .unknown) { params in
+                b.definePlainFunction(withSignature: [] => .unknown) { _ in
+                }
+            }
+        }
+
+        let original = b.finalize()
+        
+        // Splicing at CallSuperConstructor
+        b.splice(from: original, at: original.code.lastInstruction.index - 5, activeContext: [.script])
+        var actualSplice = b.finalize()
+
+        // No instructions spliced
+        XCTAssertEqual(actualSplice, Program.init())
+
+        superclass = b.defineClass() { cls in
+            cls.defineConstructor(withParameters: [.integer]) { params in
+            }
+        }
+        b.defineClass(withSuperclass: superclass) { cls in
+            cls.defineConstructor(withParameters: [.string]) { _ in
+                b.splice(from: original, at: original.code.lastInstruction.index - 5, activeContext: [.classDefinition, .function])
+            }
+        }
+        actualSplice = b.finalize()
+        
+        superclass = b.defineClass() { cls in
+            cls.defineConstructor(withParameters: [.integer]) { params in
+            }
+        }
+
+        b.defineClass(withSuperclass: superclass) { cls in
+            cls.defineConstructor(withParameters: [.string]) { _ in
+                let v0 = b.loadInt(42)
+                let v1 = b.createObject(with: ["foo": v0])
+                b.callSuperConstructor(withArgs: [v1])
+            }
+        }
+        let expectedSplice = b.finalize()
+
+        XCTAssertEqual(actualSplice, expectedSplice)
+    }
+
+    func testAsyncGeneratorSplicing() {
+        let fuzzer = makeMockFuzzer()
+        let b = fuzzer.makeBuilder()
+        b.mode = .conservative
+
+        b.defineAsyncGeneratorFunction(withSignature: FunctionSignature(withParameterCount: 2)) { _ in
+            let v3 = b.loadInt(0)
+            let v4 = b.loadInt(2)
+            let v5 = b.loadInt(1)
+            b.forLoop(v3, .lessThan, v4, .Add, v5) { _ in
+                let v0 = b.loadInt(42)
+                let _ = b.createObject(with: ["foo": v0])
+                b.await(value: v3)
+                let v8 = b.loadInt(1337)
+                b.yield(value: v8)
+            }
+            b.doReturn(value: v4)
+        }
+
+        let original = b.finalize()
+
+        b.splice(from: original, at: original.code.lastInstruction.index - 5, activeContext: [.script])
```

Comment:
Nit: would be nice to include a small comment what instruction something like `original.code.lastInstruction.index - 5` corresponds to